tool/live-monitor.js

```diff
@@ -0,0 +1,166 @@
+            return true;
+        default:
+            return false;
+    }
+}
+
+const commands = getParsedCommands();
+
+const web3Client = new Web3(networkParser(commands.network));
+
+let currentBlockNumber = commands.fromBlock;
+
+let notified = false;
+
+async function monitor() {
+
+    try {
+
+        const latestBlockNumber = await web3Client.eth.getBlockNumber();
+
+        if(currentBlockNumber === 'latest') {
+            currentBlockNumber = latestBlockNumber;
+        }
+
+        if(latestBlockNumber < currentBlockNumber) {
+            if(!notified) {
+                console.warn("latestBlockNumber was reached!");
+                notified = true;
+            }
+            return;
+        }
+
+        notified = false;
+
+        const block = await web3Client.eth.getBlock(currentBlockNumber, true);
+
+        console.info("Checking block: ", block.number);
+
+        currentBlockNumber++;
+
+        for(const transaction of block.transactions) {
+            
+            if(transaction.to === Bridge.address) {
+
+                // Requested only pegouts, if tx is not a pegout then continue
+                if(!commands.pegin && commands.pegout && !isAPegoutRelatedTransactionData(transaction.input)) {
+                    continue;
+                // Requested only pegins, if tx is not a pegin then return
+                } else if(!commands.pegout && commands.pegin && !isAPeginRelatedTransactionData(transaction.input)) {
+                    continue;
+                // Requested only pegins and pegouts, if tx is not a pegin or pegout then return
+                } else if(commands.pegin && commands.pegout && (!isAPeginRelatedTransactionData(transaction.input) || !isAPegoutRelatedTransactionData(transaction.input))) {
+                    continue;
+                }
+                // Showing all bridge events by default if commands.pegin and commands.pegout where not specified
+        
+                const rskTx = await getBridgeTransactionByTxHash(web3Client, transaction.hash);
+
+                if(commands.methods.length > 0 && !commands.methods.includes(rskTx.method.name)) {
+                    continue;
+                }
+
+                const containsAtLeast1RequestedEvent = rskTx.events.some(event => commands.events.includes(event.name));
+                if(commands.events.length > 0 && containsAtLeast1RequestedEvent) {
+                    continue;
+                }
+
+                const bridgeTxDetails = {
+                    txHash: transaction.hash,
+                    blockHash: transaction.blockHash,
+                    blockNumber: transaction.blockNumber,
+                    from: transaction.from,
+                    to: transaction.to,
+                    method: rskTx.method,
+                    events: rskTx.events
+                };
+
+                console.info("Found a tx:");
+
+                console.info(util.inspect(bridgeTxDetails, {depth: null, colors: true}));
```

Comment:
What does `util.inspect` do? Can it be replaced simply by JSON.stringify?

Reply:
It prints all the details of an object without hiding anything like a regular print does and with colors. JSON.stringify would print a single line as a string, would be really hard to read.

Reply:
For example, with a simple `console.info(bridgeTxDetails)`, the output would look like this:

```js
{
  txHash: '0xd25cb28e0390e95f8b35e846fd45555670e4721bc36ec16459454d43a6e3dd4c',
  blockHash: '0x20ff7cf9f5decfbeff2ed1d9b609ffa049fa9cc7591ab8e5a67780b377a07487',
  blockNumber: 3574973,
  from: '0xD478f3CE39cc5957b890C09EFE709AC7d4c282F8',
  to: '0x0000000000000000000000000000000001000006',
  method: BridgeMethod {
    name: 'updateCollections',
    signature: '0x0c5a9990',
    arguments: {}
  },
  events: [
    BridgeEvent {
      name: 'update_collections',
      signature: '0x1069152f4f916cbf155ee32a695d92258481944edb5b6fd649718fc1b43e515e',
      arguments: [Object]
    }
  ]
}
```
Notice the `arguments: [Object]`, this is an important information that we are not able to see by default. To see all the details, we use `util.inspect` with the `{depth: null, colors: true}` config.

With ` console.info(util.inspect(bridgeTxDetails, {depth: null, colors: true}));`, the output looks like:

```js
{
  txHash: '0xd25cb28e0390e95f8b35e846fd45555670e4721bc36ec16459454d43a6e3dd4c',
  blockHash: '0x20ff7cf9f5decfbeff2ed1d9b609ffa049fa9cc7591ab8e5a67780b377a07487',
  blockNumber: 3574973,
  from: '0xD478f3CE39cc5957b890C09EFE709AC7d4c282F8',
  to: '0x0000000000000000000000000000000001000006',
  method: BridgeMethod {
    name: 'updateCollections',
    signature: '0x0c5a9990',
    arguments: {}
  },
  events: [
    BridgeEvent {
      name: 'update_collections',
      signature: '0x1069152f4f916cbf155ee32a695d92258481944edb5b6fd649718fc1b43e515e',
      arguments: { sender: '0xD478f3CE39cc5957b890C09EFE709AC7d4c282F8' }
    }
  ]
}
```

Notice now how the `arguments` shows its data.

Reply:
I like it

Reply:
If you use `JSON.stringify(data, null, 2)` I believe it formats it as a porper JSON instead of a single line. No colors though :)

Thanks for the clarification